internal/otelcollector/config/metric/agent/receivers.go

```diff
@@ -138,3 +142,66 @@ func makePrometheusAppPodsConfig() *PrometheusReceiver {
 		},
 	}
 }
+
+func makePrometheusIstioConfig() *PrometheusReceiver {
+	return &PrometheusReceiver{
+		Config: promconfig.Config{
+			ScrapeConfigs: []*promconfig.ScrapeConfig{
+				{
+					JobName:     "istio-proxy",
+					MetricsPath: "/stats/prometheus",
+					ServiceDiscoveryConfigs: []promdiscovery.Config{
+						&promk8sdiscovery.SDConfig{
+							Role:             promk8sdiscovery.RoleEndpoint,
```

Comment:
I think we should do a pod based discovery and not make it based on endpoint based discovery. With that we will discover also pods not being behind a service.

Reply:
discovery mode switched to pod role